x/stakeibc/keeper/icacallbacks_claim.go

```diff
@@ -100,7 +100,7 @@ func (k Keeper) DecrementHostZoneUnbonding(ctx sdk.Context, userRedemptionRecord
 	}
 
 	// decrement the hzu by the amount claimed
-	hostZoneUnbonding.NativeTokenAmount = hostZoneUnbonding.NativeTokenAmount.Sub(userRedemptionRecord.NativeTokenAmount)
+	hostZoneUnbonding.ClaimableNativeTokens = hostZoneUnbonding.ClaimableNativeTokens.Sub(userRedemptionRecord.NativeTokenAmount)
```

Comment:
So the big question I am still trying to grasp to see the whole picture is how this number gets accurately updated for a partial unbonding which has to be retried at different redemption rates when the the two parts succeed.  My current understanding is that the first partial success happens at RR1, then the hostZoneUnbonding record gets picked up again from the RETRY_QUEUE when we are now at RR2, because we haven't yet assigned which users get which tokens, I think the update user records would modify all of these numbers so that the expected native amounts would be in line with RR2 right?  So we end up short tokens overall because RR2 > RR1?

If that is correct, you can still fix this issue by recomputing the implied RR for the whole hostZoneUnbonding record at the time when it finishes all of its parts successfully if you know the total stTokens it was tracking and you know the total native stake which got unbonded at the different times, then you proportionally hand out those realized unbonded tokens proportional to the stTokens in the connected user records.

If I'm off in my understanding can you walk through how this per user native token amount would be correct at this point for a partial retry at different Redemption Rates?

Reply:
To your first questions: the host zone unbonding is not updated with the latest redemption rate when it retries - it just uses whatever the RR was when the unbonding was first attempted

But this whole PR is downsteam of all that. At this point we can assume that we've unbonded all tokens for the record, _and_ that the unbonding has finished. So we can just use set the `ClaimableNativeTokens` to the records original `NativeTokenAmount` (that was set in the "refresh" function before the unbonding)

For additional context, we previously decremented `NativeTokenAmount` to track claims and then when an unbonding record finally had 0 native tokens, we would remove the record (since all users claimed). This is functionally the exact same except the field has been renamed to make it more explicit

Reply:
discussed offline, will revisit after the audit